app/models/email_export.rb

```diff
@@ -0,0 +1,69 @@
+# frozen_string_literal: true
+#
+# Service / form object for email exports
+#
+# @attr draw_id [Integer] the draw_id to scope to, if any; set to zero for
+#   drawless groups only
+# @attr size [Integer,Nil] the size to scope to, if any
+# @attr locked [Boolean] whether or not to scope to locked groups only
+class EmailExport
+  include ActiveModel::Model
+
+  attr_accessor :draw_id, :size, :locked
+  attr_reader :leaders
+
+  validates :draw_id, numericality: { only_integer: true, allow_nil: true,
+                                      greater_than_or_equal_to: 0 }
+  validates :size, numericality: { only_integer: true, allow_nil: true,
+                                   greater_than: 0 }
+  validates :locked, inclusion: { in: [true, false] }
+
+  # Initialize a new EmailExport, nillifies empty strings for draw_id and
+  # converts locked to actual booleans
+  def initialize(*args)
+    super(*args)
+    process_locked
+    process_draw_id
+  end
+
+  # Generate the e-mail export, collects the relevant groups and their leaders
+  #
+  # @return [EmailExport] the email export object, will have a leaders attribute
+  #   populated with the query results if it succeeded
+  def generate
+    execute_query if valid?
+    self
+  end
+
+  # Determine whether or not the query was scoped to a specific draw or to
+  # drawless groups
+  #
+  # @return [Boolean] whether or not the export was scoped to draw_id
+  def draw_scoped?
+    draw_scope || false
+  end
+
+  private
+
+  attr_reader :draw_scope
+
+  def process_locked
+    self.locked = locked.to_i == 1 ? true : false
+  end
+
+  def process_draw_id
+    return unless draw_id.present?
+    @draw_scope = true
+    # this kinda makes the numericality validations on :draw_id useless, not
+    # sure if it's worth keeping them or not.
+    self.draw_id = nil if draw_id.to_i.zero?
```

Comment:
This is used to scope for drawless groups - we need to set it to `nil` from `0` but differentiate that from the case where no `draw_id` is passed (in which case we don't scope by draw). However, if you pass _any_ value other than a positive integer in, `to_i` will turn it into zero regardless. In other words, if you param hack to send `draw_id = 'a'`, that will still scope to drawless groups.

That said, I now disagree with my comment - the numericality validation on `draw_id` is _not_ useless since it prevents the queries from borking if someone called `execute_query` after assigning some non-integer value to `draw_id` (resulting in an invalid query). I'm going to remove the comment (but copy it below for posterity):

```ruby
    # this kinda makes the numericality validations on :draw_id useless, not
    # sure if it's worth keeping them or not.
```